### DISCUSSION TITLE
External formatter: Cannot run program "/usr/bin/pg_format": error=13, Permission denied

### DISCUSSION THREAD

**Discussion by TheMechanist902** [dbeaver/dbeaver#18972] [Q&A]
Hi everyone,

I'm trying to get the pg_format CLI formatter to work ob Ubuntu 22.04 LTS (Installed via apt-get) as an external formatter. I used this guide here as reference: https://tanutaran.medium.com/format-postgresql-beautifully-on-dbeaver-with-pg-formatter-external-formatter-c91a13aad5cd

But it doesn't work. Even with a full path given by the "which" CLI command I always get an error that I assume is from missing rights on that command:
![image](https://user-images.githubusercontent.com/8171872/217064802-6c496499-5468-4295-a5f6-57d8c56c12f9.png)
It should be noted that this error occurs even if linking to files that don't exist. Just to clarify: This error occurs in the current .deb packages installed DBeaver as well as in the one from the snap. Usually I use the snap version for automated updates, but I initially thought the snap sandbox may have blocked program access.

Can somebody help me out? Maybe I'm not using the correct syntax? I would not like to change access rights outside of my home folder if avoidable. But if that is the only option of course I would consider it.
Anyway thanks for your help and have a nice day.

**Comment by LonwoLonwo** [comment]
You can find the existing issue here: https://github.com/dbeaver/dbeaver/issues/18958

**Reply by TheMechanist902** [reply]
Thank you for the promptly response. 'm looking forward for a fix in the next release. Have a great day.

**Reply by LonwoLonwo** [reply]
We are working on it!
